scripts/update-sidecar.ps1

```diff
@@ -19,44 +19,49 @@ $RootDir = Split-Path $PSScriptRoot -Parent
 $SrcDir = Join-Path $RootDir "src"
 
 $OutputDir = Join-Path $RootDir "src" "generated"
+
+# Remove the linux and linuxDiagTrace directories - so that we delete files that have been abandoned
 if (Test-Path $OutputDir) {
-    Remove-Item $OutputDir -Recurse -Force -Exclude 'CMakeLists.txt'
-}
 
-$Files = [System.Collections.Generic.List[string]](Get-ChildItem -Path "$SrcDir\*" -Recurse -Include *.c,*.h,*.cpp,*.hpp -File)
-$Files.Sort([SimpleStringComparer]::new())
+    if (Test-Path (Join-Path $OutputDir linux)) {
+        Remove-Item (Join-Path $OutputDir linux) -Recurse -Force
+        Remove-Item (Join-Path $OutputDir linuxDiagTrace) -Recurse -Force
+    }
+}
 
 $Sidecar = Join-Path $SrcDir "manifest" "clog.sidecar"
 $ConfigFile = Join-Path $SrcDir "manifest" "msquic.clog_config"
 
 $TmpOutputDir = Join-Path $RootDir "build" "tmp"
+$ClogDir = Join-Path $RootDir "build" "clog"
 
+# Create directories
 New-Item -Path $OutputDir -ItemType Directory -Force | Out-Null
+New-Item -Path (Join-Path $OutputDir linux) -ItemType Directory -Force | Out-Null
+New-Item -Path (Join-Path $OutputDir linuxDiagTrace) -ItemType Directory -Force | Out-Null
 
-Remove-Item $Sidecar -Force -ErrorAction Ignore | Out-Null
+# Build CLOG, placing results into the CLOG directory under our build directory
+dotnet publish ../submodules/clog/src/clog -o ${ClogDir} -f net5.0
 
-clog --installDirectory (Join-Path $OutputDir common)
+#
+# You may be tempted to delete the sidecar - DO NOT DO THIS - the sidecare
+#     exists for several purposes - one of those is to verify signatures of trace calls have
+#     not changed.  If you delete the sidecar, you'll miss errors that may save you broken contracts
+#     that occur when you decode
```

Comment:
```suggestion
# You may be tempted to delete the sidecar - DO NOT DO THIS - the sidecare
#  exists for several purposes - one of those is to verify signatures of trace calls have
#  not changed.  If you delete the sidecar, you'll miss errors that may save you broken contracts
#  that occur when you decode
```
I agree with this so long as the following are not considering changes:

1. Moving an event around, relative to other events; including adding a new event before an existing one or moving between files.
2. Changing the human readable text. We tweak strings occasionally and want to replace existing event definitions; not add new ones.

Reply:
I like;  I'm required to disagree with #2, but that's ivory tower.  I'd argue you should create a new event if you're changing the structure of the event but at the end of the day thats your call

Reply:
Yeah. Makes sense, but practically, I think we should just consider the binary encoding the "signature", not the string used to display that data. In other words, we should just ensure tools that decode the binary data continue to work.